### PR TITLE
Combat: implement the infrastructure to change the unit's "main" cell when positioning it on the battlefield

### PR DESCRIPTION
**Pull request by oleg-derevenetz**
Potential prerequisite to #5048

Now an attack can (optionally) be performed from the tail cell of the wide unit (see discussion in the #5048). How this may work (currently this behavior is disabled):

https://user-images.githubusercontent.com/32623900/159389474-e79606e1-ba84-4eaa-8e48-1f93319e23e1.mp4

This allows to implement OG-like direct attacks in future (see `battle_interface.cpp` for some examples).

### CODE REVIEW

File: src/fheroes2/battle/battle_action.cpp [hunk 1]
```diff
@@ -1171,6 +1177,9 @@ void Battle::Arena::ApplyActionSpellMirrorImage( Command & cmd )
         }
     }
     else {
-        DEBUG_LOG( DBG_BATTLE, DBG_WARN, "spell: " << Spell( Spell::MIRRORIMAGE ).GetName() << " false" );
+        DEBUG_LOG( DBG_BATTLE, DBG_WARN,
+                   "incorrect param"
+                       << ": "
+                       << "who: " << who );
```
Comment (ihhub) [inline src/fheroes2/battle/battle_action.cpp:1183]: These 3 strings can be as a single one or something is missing here.

File: src/fheroes2/battle/battle_action.cpp [hunk 2]
```diff
@@ -1072,17 +1074,19 @@ void Battle::Arena::ApplyActionSpellTeleport( Command & cmd )
...
 
-        DEBUG_LOG( DBG_BATTLE, DBG_TRACE, "spell: " << spell.GetName() << ", src: " << src << ", dst: " << dst );
+        DEBUG_LOG( DBG_BATTLE, DBG_TRACE, "src: " << src << ", dst: " << dst );
     }
     else {
-        DEBUG_LOG( DBG_BATTLE, DBG_WARN, "spell: " << spell.GetName() << " false" );
+        DEBUG_LOG( DBG_BATTLE, DBG_WARN,
+                   "incorrect param"
+                       << ": "
+                       << "src: " << src << ", dst: " << dst );
```
Comment (ihhub) [inline src/fheroes2/battle/battle_action.cpp:1089]: First 3 strings can be combined into one or something is missing here.

**Review by ihhub — COMMENTED** [review]
Hi @oleg-derevenetz , I left just 2 comments related to logging. Could you please take a look?

File: src/fheroes2/battle/battle_action.cpp (hunk 1 above)
Comment (oleg-derevenetz) [reply src/fheroes2/battle/battle_action.cpp]: Hi @ihhub that's due to some funny behavior of `clang-format`. It refuses to place them on the single line. At least I combined the first two strings into one.

Comment (oleg-derevenetz) [reply src/fheroes2/battle/battle_action.cpp]: Hi @ihhub This is a kind of `DEBUG_LOG` output style in the `battle_action.cpp`. I simplified it a bit.

**Comment by ihhub** [comment]
@oleg-derevenetz , thanks a lot for these changes!

**Pull request merged**